spec/openai/client/http_spec.rb

```diff
@@ -134,44 +139,52 @@
         it "does not call the user proc" do
           bad_examples.each do |chunk|
             expect(user_proc).to_not receive(:call)
-            stream.call(chunk)
+            stream.call("#{chunk}\n\n")
           end
         end
       end
 
       context "when called with a string containing that looks like a JSON object but is invalid" do
-        let(:chunk) do
-          <<-CHUNK
-            data: { "foo": "bar" }
-            data: { BAD ]:-> JSON }
-          CHUNK
-        end
-
         it "does not raise an error" do
           expect(user_proc).to receive(:call).with(JSON.parse('{"foo": "bar"}'))
 
           expect do
-            stream.call(chunk)
+            stream.call(<<~CHUNK)
+              data: { "foo": "bar" }
+
+              data: { BAD ]:-> JSON }
+
+              #
+            CHUNK
           end.not_to raise_error
         end
       end
 
       context "when called with a string containing an error" do
-        let(:chunk) do
-          <<-CHUNK
-            data: { "foo": "bar" }
-            error: { "message": "A bad thing has happened!" }
-          CHUNK
-        end
-
         it "does not raise an error" do
           expect(user_proc).to receive(:call).with(JSON.parse('{ "foo": "bar" }'))
           expect(user_proc).to receive(:call).with(
             JSON.parse('{ "message": "A bad thing has happened!" }')
           )
 
           expect do
-            stream.call(chunk)
+            stream.call(<<~CHUNK)
+              data: { "foo": "bar" }
+
+              error: { "message": "A bad thing has happened!" }
```

Comment:
AFAIK, there's no such thing as an "error" field. But that is beyond the scope of this PR.